generic3g/vertical/VerticalLinearMap.F90

```diff
@@ -0,0 +1,121 @@
+#include "MAPL_Generic.h"
+
+module mapl3g_VerticalLinearMap
+
+   use mapl_ErrorHandling
+   use mapl3g_CSR_SparseMatrix, only: SparseMatrix_sp => CSR_SparseMatrix_sp
+   use mapl3g_CSR_SparseMatrix, only: add_row
+   use mapl3g_CSR_SparseMatrix, only: sparse_matmul_sp => matmul
+   use, intrinsic :: iso_fortran_env, only: REAL32
+
+   implicit none
+   private
+
+   public :: compute_linear_map_fixedlevels_to_fixedlevels
+   public :: apply_linear_map
+
+   type IndexValuePair
+      integer :: index
+      real(REAL32) :: value_
+   end type IndexValuePair
+
+   interface operator(==)
+      procedure equal_to
+   end interface operator(==)
+
+contains
+
+   subroutine apply_linear_map(matrix, fin, fout)
+      ! real(REAL32), intent(in) :: matrix(:, :)
+      type(SparseMatrix_sp) :: matrix
+      real(REAL32), intent(in) :: fin(:)
+      real(REAL32), allocatable, intent(out) :: fout(:)
+
+      fout = sparse_matmul_sp(matrix, fin)
```

Comment:
This is not a wrapper, but a renaming to differentiate from the intrinsic `matmul`
```Fortran
use mapl3g_CSR_SparseMatrix, only: sparse_matmul_sp => matmul
```

Reply:
But ... I named it matmul() on purpose, as it is intended to be used in the same manner ...

we can discuss at tomorrows standup.  Or maybe it will be more clear to me on my 2nd review.

Reply:
Removed `apply_linear_map` wrapper.